### ISSUE TITLE
No predictions

### ISSUE THREAD

**Issue by Chase-Nolan** [ultralytics/ultralytics#1749]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hey, 

So I'm training with 480 Images, all with labels, and validating with 120. I heard that Yolo works with 640 by 640 images but mine are 1920 by 1080.

- ### **Here is the python code**

`**Yolov8s training object detection model**
from ultralytics import YOLO

**Load a model**
model = YOLO("yolov8s.yaml")  # build a new model from scratch

**Use the model and train it on Nvidia GPU, use an image size of 1920x1080, and train for 1 epoch**
if __name__ == '__main__':
    results = model.train(data="config.yaml", epochs=10, device=0)  # train the model`

After 10 epochs the val_predictions have nothing at all. Could the image size be the reason?

- ### **This is my config.yaml file**

`**Path to the YOLOv8-Detection folder**
path: C:\Users\Chase\Desktop\YoloV8-Detection

**Directories to the training and validation sets**
train: C:\Users\Chase\Desktop\YoloV8-Detection\images\train
val: C:\Users\Chase\Desktop\YoloV8-Detection\images\val

**Input size of the images**
model:
  type: YOLOv8
  input_size:
    width: 1920
    height: 1080

**Classes**
names:
  0: Body
  1: Head
  2: Player`

**Here are the results:**

![confusion_matrix](https://user-images.githubusercontent.com/99596659/229276922-2bf4e40b-3ec0-4c3a-bc99-f96e47113cc7.png)
![F1_curve](https://user-images.githubusercontent.com/99596659/229276940-0ef096be-d5f8-438c-8516-bd1e2b6b7308.png)
![labels_correlogram](https://user-images.githubusercontent.com/99596659/229276958-b2723a25-8fb6-4c32-b70b-5ee32ff00083.jpg)
![P_curve](https://user-images.githubusercontent.com/99596659/229276968-da82c150-1e48-4616-94ef-28ecf61c03cb.png)
![PR_curve](https://user-images.githubusercontent.com/995966
![results](https://user-images.githubusercontent.com/99596659/229276991-1403c9e5-4466-4243-9637-d5d09aff0eb0.png)
59/229276974-e9077380-df88-4804-bb8e-da29f78aae35.png)
![R_curve](https://user-images.githubusercontent.com/99596659/229276981-296ab235-9ced-4047-af86-80fbe8859d4e.png)

![labels](https://user-images.githubusercontent.com/99596659/229276950-a0a144a6-ff1a-43d2-9913-27f89ef5afb8.jpg)

**Comment by glenn-jocher** [comment]
@Chase-Nolan if your custom-trained YOLOv8 model is not detecting anything after a few epochs, there might be several reasons behind this. Here are some suggestions to improve your model's performance:

Increase the dataset size: Try to gather more labeled data or use data augmentation techniques to increase the size of your dataset. A larger dataset can improve the model's ability to detect objects.

Fine-tune a pre-trained model: Instead of training the model from scratch, you can start with a pre-trained YOLOv8 model and fine-tune it on your dataset. This approach allows the model to benefit from pre-existing knowledge and may result in faster convergence and better performance.

Adjust learning rate and other hyperparameters: Experiment with different learning rates, batch sizes, and other hyperparameters to find a better combination that improves your model's performance.

Increase the number of training epochs: Sometimes, a few epochs might not be enough for the model to learn the features of the dataset. Increase the number of epochs and monitor the performance on the validation dataset to prevent overfitting.

Check annotation quality: Ensure that the annotations for your dataset are accurate and correctly formatted. Incorrect or mislabeled data can negatively impact the model's performance.

Evaluate model performance: Make sure you are evaluating your model's performance using appropriate metrics like mean Average Precision (mAP). This will help you understand how well your model is performing on your dataset.

Visualize intermediate results: Visualize the results at different stages of the training process to get an idea of what the model is learning. This can help you identify issues with the training process or dataset.

Check the loss function: Make sure you are using an appropriate loss function for the task. In the case of YOLOv8, the loss function should be suitable for object detection.

By carefully analyzing and addressing these aspects, you can improve the performance of your custom YOLOv8 model on your dataset.

**Comment by Chase-Nolan** [comment]
Thanks for the response. It turns out that I just had to increase my dataset. Although it produces results, they arent amazing and i think its because I am trying to detect far away objects. I only have a 3060ti (8gm vram) so I can only train at the max of imgsz=864, I do have a 1070 but I cant get multigpu to work so is there anything you can suggest I do to be able to train my model on 1280 pixels instead of 864 without running our if vram?

**Comment by Chase-Nolan** [comment]
Hey, thanks for the response. I can't find anything on the docs for Gradient accumulation? is that a setting to pass into 

`model.train(data="config.yaml", epochs=100, imgsz=1280, device=0, save_period=15, batch=8)`

Currently I have batch at 8, I think the usual batch is 16 so hopefully this improves it.

**Comment by glenn-jocher** [comment]
@Chase-Nolan, Gradient accumulation is usually implemented in the training loop rather than as a setting. It involves accumulating gradients from multiple batches before doing a weight update, which effectively increases the effective batch size. In YOLOv8, you may implement gradient accumulation by updating the model weights every few batches instead of every batch. Start by setting the batch size at a lower value that fits in your GPU, like 8, and increase the number of batches before calling the `model.step()` method, which updates the weights. This Can be done in the following way, for instance:
```
for i, (imgs, targets, paths, _) in enumerate(dataloader):
    loss, outputs = model(imgs.to(device), targets.to(device))
    loss.backward()

    if (i + 1) % accumulation_steps == 0:
        optimizer.step()
        optimizer.zero_grad()
```

As for the Docker image, it can be used to train with two GPUs by running the training process inside the Docker container, which is an isolated environment that doesn't interfere with your host machine. This way, you can run the training process using one GPU inside the container and the other GPU in the host machine. You can also use the Docker image to ensure that you have all the required dependencies installed for the training process without affecting your host operating system.

**Comment by glenn-jocher** [comment]
@Chase-Nolan sure, I'd be happy to explain the concept behind using Docker to train on multiple GPUs.

Docker is a tool that can create and run containers, which are isolated environments to run processes. By using Docker, you can set up an environment with all the necessary dependencies and configurations to train your YOLOv8 model. With support for GPU configurations, Docker can be used to train a model on multiple GPUs simultaneously.

When using Docker to train on multiple GPUs, you'll create a container that runs your training process. You can then choose to configure Docker to use multiple GPUs; this can be done by passing device IDs to the Docker container when running it, `NVIDIA_VISIBLE_DEVICES`. This configuration will enable YOLOv8 to train using both GPUs.

Overall, using Docker for training on multiple GPUs comes down to creating the Docker container, passing the correct commands and configurations to ensure that the container uses both GPUs for training, and then running the training process inside the container.
